### ISSUE TITLE
SameActualExpectedValue (NUnit2009) should not be a warning

### ISSUE THREAD

**Issue by verdie-g** [nunit/nunit.analyzers#842]
[NUnit2009](https://docs.nunit.org/articles/nunit-analyzers/NUnit2009.html) warns for this code
```csharp
MyClass x = new()
Assert.That(x, Is.EqualTo(x));
```

but when testing an equals method, that is totally acceptable. Maybe this rule should be restrained to a bunch of known .NET types?

**Comment by manfred-brands** [comment]
In most cases the above code is an error.

In the case where you want to test _your_ `Equals` implementation, you should not use `Is.EqualTo` at all as your method won't even get called because Nunit will first do a `ReferenceEquals`.

Instead you should use code like:
```csharp
#pragma warning disable NUnit2010 // Use EqualConstraint for better assertion messages in case of failure

Assert.That(instance.Equals(instance), Is.True, "Equals with self");
Assert.That(instance.Equals(same), Is.True, "Equals with same");
Assert.That(same.Equals(instance), Is.True, "Equals with same reversed");

Assert.That(instance.Equals(different), Is.False, "Equals with different");
Assert.That(different.Equals(instance), Is.False, "Equals with different reversed");

#pragma warning restore NUnit2010 // Use EqualConstraint for better assertion messages in case of failure
```

Where `instance` and `same` are different instances of the class created with the same parameters and `different` uses other constructor parameters.

**Comment by verdie-g** [comment]
> you should not use Is.EqualTo at all as your method won't even get called because Nunit will first do a ReferenceEquals.

Indeed. Thanks!
